File: gospider/examples/legacy/cluster_worker.go
package main

import (
	"gospider/spider"
	"os"
)

func clusterWorkerMain() {
	masterURL := "http://localhost:5000"
	if len(os.Args) > 1 {
		masterURL = os.Args[1]
	}

	worker := spider.NewClusterWorker(masterURL)
	worker.Run()
}

File: rustspider/examples/legacy/cluster_worker.rs
use rustspider::DistributedWorker;
use std::env;

#[tokio::main]
async fn main() -> anyhow::Result<()> {
    let master_url = env::args()
        .nth(1)
        .unwrap_or_else(|| "http://localhost:5000".to_string());

    let worker = DistributedWorker::new(&master_url, 4);
    worker.run().await;
    Ok(())
}
